### ISSUE TITLE
Supported config file options

### ISSUE THREAD

**Issue by skitzo2000** [MicrosoftDocs/windowsserverdocs#3717]
I've searched the documentation and cannot find any references to available options for the SSHClient config file.  More specifically I wanted to try using an "Include" statement as I manage a ton of servers via SSH and I like to keep a tidy config file.  The Include statement makes that much easier.  I tried and it doesn't seem to work.  But some documentation on what options are allowed would be nice.

**Comment by illfated** [comment]
Which docs.microsoft.com page is this based on? Would you be able to re-post your issue ticket directly on the page in question, making it easier to reference the documentation page in need of improvements?

**Comment by skitzo2000** [comment]
That's the problem.  The config file isn't mentioned anywhere in the documentation.

**Comment by illfated** [comment]
Fair enough, good point. Do you have any opinion with regards to which page the missing documentation would belong in, or do you think the topic is big enough for its own page?

**Comment by RAJU2529** [comment]
@skitzo2000 . I found .ssh config  files are astored  under 
C:\Program Files (x86)\Git\etc\ssh\
moduli
ssh_config
sshd_config

for more information, visit official websites
**https://www.ssh.com/ssh/config**
**https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh**

Note:: I am not familiar with SSH

**Comment by illfated** [comment]
@RAJU2529 : It looks like your example is specific to Git (local client for connecting to GitHub).

In this case, it is more about the Microsoft Docs pages:
- https://docs.microsoft.com/windows-server/administration/openssh/openssh_overview
- https://docs.microsoft.com/windows-server/administration/openssh/openssh_install_firstuse
- https://docs.microsoft.com/windows-server/administration/openssh/openssh_server_configuration
- https://docs.microsoft.com/windows-server/administration/openssh/openssh_keymanagement

**Comment by RAJU2529** [comment]
@skitzo2000. Please follow  @illfated suggested Microsoft docs websites.

Thanks  for sharing the websites

**Comment by illfated** [comment]
You are welcome. Even so, I don't think we are any closer to answering the original question of **available options for the SSHClient config file**.

**Comment by skitzo2000** [comment]
@RAJU2529 The referenced pages don't contain any information regarding config files which is what I was asking about.  I'm very familiar with SSH and its options as I use it daily to work with Linux servers.

@illfated I'm not sure if it deserves its own page or not since I'm not sure how many options MS built in for the config file.  But it would be nice to have some documentation.  Most normal SSH options seem to be supported just not the Include option I was hoping for.

**Comment by illfated** [comment]
Very well, from what I can see in the MS Docs pages, the author is listed as @maertendMSFT , maybe we could have a comment from Danny about this.

Then again, there is this sentence in the MS Docs configuration page, https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_server_configuration :
> OpenSSH maintains detailed documentation for configuration options online at [OpenSSH.com](https://www.openssh.com/manual.html), which is not be duplicated in this documentation set.

@skitzo2000 : I presume you have already read most of that, but just to make sure, have you found anything related to this issue in the pages underneath https://www.openssh.com/manual.html ?

**Comment by skitzo2000** [comment]
@illfated @maertendMSFT I definitely have read those pages, The referenced page on the MS side is for the OpenSSH server config (sshd_config).  Not the client config for ssh.

It is good to note that on the OpenSSH documentation there is an entire page dedicated to the client config options here: https://man.openbsd.org/ssh
Maybe that could be used as a reference point for the MS documentation.

**Comment by dknappettmsft** [comment]
#assign:dknappettmsft

**Comment by dknappettmsft** [comment]
Hi @skitzo2000, I have added a paragraph to the '[Get started with OpenSSH](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse)' article to include the client config locations in Windows.  This change should be visible later today.

For what to include in those config files, the reference really is the OpenSSH manual and we don't duplicate this in the Microsoft documentation.

I have tested the Include functionality in Windows and it works as detailed in the OpenSSH manual - Windows doesn't do this differently. As an example, I created a test config file in `C:\Users\User\.ssh` called `test_config`. In that file I had this content:

```
Host entry1
  HostName 10.0.0.11
  User user

Host entry2
  HostName 10.0.0.12
  User user

Host entry3
  HostName 10.0.0.13
  User user
```

I can run the command `ssh -F "C:\Users\User\.ssh\test_config" entry1` and it will connect to entry1 in that config file. If I also add another file called `config `in `C:\Users\User\.ssh` and add the line Include `C:\Users\User\.ssh\test_config`, I can then run the shorter command `ssh entry1`.

I hope this helps.
